rpc/wrtc_signaling_server.go

```diff
@@ -164,16 +164,13 @@ func (srv *WebRTCSignalingServer) Call(req *webrtcpb.CallRequest, server webrtcp
 		}
 
 		srv.callMu.RLock()
-		// we assume the number of goroutines is bounded by the gRPC server invoking this method.
-		srv.callWorkers.Add(1)
-		srv.callMu.RUnlock()
-
-		// Check if cancelCtx has errored: underlying server may have been
-		// `Close`ed, in which case we mark this call worker as `Done` and return.
+		// Atomically check if the cancelCtx was canceled, and if not, add to the `callWorkers`.
 		if err := srv.cancelCtx.Err(); err != nil {
-			srv.callWorkers.Done()
+			srv.callMu.RUnlock()
```

Comment:
Based on my internal model, this wasn't technically a bug (and why that is bothers me that I don't have a great way of describing the meat of the issue). But figured it was worth making it all look the same.